File: web/Workshop_web01/WebContent/js/book.js
function check() {
	var form = document.Frm;
	
	if((form.isbn1.value)==''||(form.isbn2.value)==''||(form.isbn2.value)==''){
		alert("도서 번호를 입력해주세요.");
		form.isbn1.focus();
		return false;
	} 
	
	if(parseInt(form.isbn1.value)>9999 && parseInt(form.isbn1.value)<1000 ){
		alert("첫번째 자리는 4자리 입력하여 주세요");
		form.isbn1.focus();
		return false;
	}else if(parseInt(form.isbn2.value)>999 && parseInt(form.isbn2.value)<100 ){
		alert("두번째 자리는 3자리 입력하여 주세요");
		form.isbn2.focus();
		return false;
	}else if(parseInt(form.isbn3.value)>999 && parseInt(form.isbn3.value)<100 ){
		alert("세번째 자리는 3자리 입력하여 주세요");
		form.isbn3.focus();
		return false;
	}
	
	if((form.title.value)==''){
		alert("도서 제목을 입력해주세요.");
		form.title.focus();
		return false;
	}
	
	if((form.kinds.value)=="종류"){
		alert("책의 종류를 선택해주세요");
		form.kinds.focus();
		return false;
	}
	
	if((form.writer.value)==''){
		alert("저자를 입력해주세요.");
		form.writer.focus();
		return false;
	}

}

File: web/Workshop_web01/src/servelt/controller/Book.java
package servelt.controller;

import java.io.IOException;
import java.sql.Date;
import java.sql.SQLException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import servelt.model.book.BookDAOimpl;
import servelt.model.book.BookVo;

public class Book extends HttpServlet {
	private static final long serialVersionUID = 1L;
	
	protected void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
		doProcess(req,res);
	}

	
	protected void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
		doProcess(req,res);
	}
	
	protected void doProcess(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
		
		Date publish_date = null;
		int price = 0;
		
		//값 받아와서
		String isbn = req.getParameter("isbn1")+"-"+req.getParameter("isbn2")+"-"+req.getParameter("isbn3");
		String title = req.getParameter("title");
		String catalogue = req.getParameter("kinds");
		String nation = req.getParameter("country");
	
		if (req.getParameter("release").equals("")) {
			publish_date = Date.valueOf("9999-12-13");
		}else {
		 publish_date = Date.valueOf(req.getParameter("release"));}
	
		String publisher = req.getParameter("publisher ");
		String author = req.getParameter("writer");
		
		if(req.getParameter("price").equals("")) {
			price = 0;
		}else {
		price = Integer.parseInt(req.getParameter("price"));}
		
		String currency = req.getParameter("unit");
		String desc = req.getParameter("desc");
		
		
		//객체생성
		BookVo book = new BookVo(isbn, title, catalogue, nation, publish_date, publisher, author, price, currency, desc);
		System.out.println(book.toString());
		
		
		//dao bizs
		BookDAOimpl dao = BookDAOimpl.getInstance();
		try {
			dao.bookRegister(book);
		} catch (SQLException e) {
			//e.printStackTrace();
			System.out.println("bookRegister SQL오류");
		}
		req.setAttribute("book", book);
		
		//forwarding
		RequestDispatcher dispatcher= req.getRequestDispatcher("book/result.jsp");
		dispatcher.forward(req, res);
	}

}
